utils/web_test.go

```diff
@@ -0,0 +1,74 @@
+package utils
+
+import (
+	"reflect"
+	"testing"
+)
+
+func TestNewWeb(t *testing.T) {
+	tests := []struct {
+		name string
+		log  Logger
+		want Web
+	}{
+		{"1", fakeLogger, &WebStruct{l: fakeLogger}},
+		{"1", nil, &WebStruct{l: nil}},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			if got := NewWeb(tt.log); !reflect.DeepEqual(got, tt.want) {
+				t.Errorf("NewWeb() = %v, want %v", got, tt.want)
+			}
+		})
+	}
+}
+
+func TestWebStruct_DownloadFile(t *testing.T) {
+	tests := []struct {
+		name    string
+		url     string
+		want    []byte
+		wantErr bool
+	}{
+		{"1", "http://mr.mzxyptlk/", nil, true},
+		{"2", "https://raw.githubusercontent.com/openSUSE/helm-mirror/master/utils/fixtures/download.txt", []byte("first test line\n"), false},
+	}
+	w := &WebStruct{l: fakeLogger}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			got, err := w.DownloadFile(tt.url)
+			if (err != nil) != tt.wantErr {
+				t.Errorf("WebStruct.DownloadFile() error = %v, wantErr %v", err, tt.wantErr)
+				return
+			}
+			if !reflect.DeepEqual(got, tt.want) {
+				t.Errorf("WebStruct.DownloadFile() = %v, want %v", got, tt.want)
+			}
+		})
+	}
+}
+
+func TestWebStruct_DownloadFileString(t *testing.T) {
+	tests := []struct {
+		name    string
+		url     string
+		want    string
+		wantErr bool
+	}{
+		{"1", "http://mr.mzxyptlk/", "", true},
+		{"3", "https://raw.githubusercontent.com/openSUSE/helm-mirror/master/utils/fixtures/download.txt", "first test line\n", false},
+	}
+	w := &WebStruct{l: fakeLogger}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			got, err := w.DownloadFileString(tt.url)
+			if (err != nil) != tt.wantErr {
+				t.Errorf("WebStruct.DownloadFileString() error = %v, wantErr %v", err, tt.wantErr)
+				return
+			}
+			if got != tt.want {
+				t.Errorf("WebStruct.DownloadFileString() = %v, want %v", got, tt.want)
```

Comment:
It's not a good idea to rely on an the availability of a remote hosted file. I think the best solution would be to start a local webserver (go has native support for that inside of its stdlib) and attempt a download from localhost.